src/javascript/SelectorTypes/SelectorTypes.onChange.js

```diff
@@ -35,31 +39,51 @@ const registerSelectorTypesOnChange = registry => {
                         value: currentValue === null ? [] : currentValue
                     });
 
-                    const {data} = await editorContext.client.query(
```

Comment:
I had to refactor this as the await in a forEach() loop is factually not used and error prone. Instead, return a list of promise, and continue when all promise have resolved.